### PR TITLE
net-proxy/3proxy: version bump to 0.8.8

### PR DESCRIPTION
**Pull request by powerman**
Package-Manager: portage-2.3.0

### CODE REVIEW

File: net-proxy/3proxy/3proxy-0.8.8.ebuild [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+SRC_URI="https://github.com/z3APA3A/3proxy/archive/${P}.tar.gz"
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~x86 ~amd64 ~ppc"
+IUSE=""
+
+S="${WORKDIR}/${PN}-${P}"
+
+src_prepare() {
+	epatch "${FILESDIR}"/${P}-gentoo.patch
```
Comment (SoapGentoo) [inline net-proxy/3proxy/3proxy-0.8.8.ebuild]: don't use `epatch` in EAPI=6, stick the patch in global `PATCHES` and let `default` do the patching, you can then get rid of the `inherit eutils`

File: net-proxy/3proxy/3proxy-0.8.8.ebuild [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	cp Makefile.Linux Makefile || die
```
Comment (SoapGentoo) [inline net-proxy/3proxy/3proxy-0.8.8.ebuild]: move this after `default`

File: net-proxy/3proxy/3proxy-0.8.8.ebuild [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+src_install() {
+	local x
+
+	cd src || die
+	dobin 3proxy
+	for x in proxy socks ftppr pop3p tcppm udppm mycrypt dighosts icqpr smtpp; do
+		newbin ${x} ${PN}-${x}
+		[[ -f "${S}"/man/${x}.8 ]] && newman "${S}"/man/${x}.8 ${PN}-${x}.8
+	done
+	cd ..
```
Comment (SoapGentoo) [inline net-proxy/3proxy/3proxy-0.8.8.ebuild]: missing `|| die`
if you're cd'ing into a dir and from it again, consider using `pushd/popd >/dev/null`

File: net-proxy/3proxy/3proxy-0.8.8.ebuild [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+	doman man/3proxy*.[38]
+
+	dodoc README
+	docinto html
+	dodoc -r doc/html/*
+	docinto cfg
+	dodoc -r cfg/*
```
Comment (SoapGentoo) [inline net-proxy/3proxy/3proxy-0.8.8.ebuild]:
```
DOCS=( README cfg )
HTML_DOCS=( doc/html/. )
einstalldocs
```
instead of the previous 5 lines

**Comment by monsieurp** [comment]
ping @powerman

**Comment by powerman** [comment]
@monsieurp I didn't forget about this and I'm going to make all requested changes, but I was ill, then have a lot of other work, so chances are I'll do this on next week.

**Comment by powerman** [comment]
l did all requested changes.

Also I'm going to release next version (0.8.9) as soon as I figure out how to save downloaded SRC_URI using alternative name, because https://github.com/z3APA3A/3proxy/archive/0.8.9.tar.gz saves as 0.8.9.tar.gz in distfiles directory and I don't think that's ok (btw, when this url is downloaded by browser file name changes to 3proxy-0.8.9.tar.gz, so maybe it's a bug in portage to ignore download file name provided in Content-Disposition HTTP header).

**Comment by powerman** [comment]
I've added 0.8.9 to this PR, but I can make separate PR with it if needed.

If you don't mind, a couple of proxy-maint newbie questions:
- what I should do to test package on arches and turn some KEYWORDS to `ARCH` from `~ARCH`?
- what is the policy to remove old versions?

**Comment by kensington** [comment]
Hi, sorry for the delay in attending to your PR.

> If you don't mind, a couple of proxy-maint newbie questions:
> 
>     what I should do to test package on arches and turn some KEYWORDS to ARCH from ~ARCH?

File a bug to request stabilisation - in general arch teams are responsible for making the change. See https://wiki.gentoo.org/wiki/Stable_request for further details.

>     what is the policy to remove old versions?

This is mostly up to the maintainer. Avoid removing the last stable version unless there's a really good reason. When doing a version bump, it's nice to leave at least one old version behind so the user can easily downgrade if there's a problem for them. It's good practice to remove old versions in their own commit.

**Comment by kensington** [comment]
> Also I'm going to release next version (0.8.9) as soon as I figure out how to save downloaded SRC_URI using alternative name, because https://github.com/z3APA3A/3proxy/archive/0.8.9.tar.gz saves as 0.8.9.tar.gz in distfiles directory and I don't think that's ok

Correct, and renaming is done like this: ``SRC_URI="https://example.com/${PV}.tar.gz -> ${P}.tar.gz"``

**Comment by kensington** [comment]
Thanks, picked as f3ff448671e8102de87143905d9b847362f15997...f802519a22365239050919537ead12ff62224624 (I squashed the 0.8.8 changes together).

**Comment by kensington** [comment]
I also noticed a couple of things that could be fixed for next time (did not block this PR because they are not regressions):

* ``KEYWORDS`` is normally sorted
* ``-g`` is passed to the compiler on several occasions - this shouldn't occur as it should be up to the user if they want debugging symbols or not
* ``gcc`` is called directly - depending on the build system behaviour this might be fixed by inheriting ``toolchain-funcs``.eclass and either calling ``tc-export CC`` or building with ``emake CC="$(tc-getCC)"``

**Comment by powerman** [comment]
> gcc is called directly - depending on the build system behaviour this might be fixed by inheriting toolchain-funcs.eclass and either calling tc-export CC or building with emake CC="$(tc-getCC)"

As far as I see - this won't happens. Direct call to gcc exists in two files https://github.com/z3APA3A/3proxy/blob/devel/src/plugins/LdapPlugin/com-unix.sh and https://github.com/z3APA3A/3proxy/blob/devel/src/plugins/LdapPlugin/com-win32.bat but it doesn't looks like these files are executed while building 3proxy. The first one even tries to compile non-existing `ldapauth_new.c`.

Other points fixed in https://github.com/gentoo/gentoo/pull/3939, thanks for review.

**Comment by kensington** [comment]
> As far as I see - this won't happens. Direct call to gcc exists in two files https://github.com/z3APA3A/3proxy/blob/devel/src/plugins/LdapPlugin/com-unix.sh and https://github.com/z3APA3A/3proxy/blob/devel/src/plugins/LdapPlugin/com-win32.bat but it doesn't looks like these files are executed while building 3proxy. The first one even tries to compile non-existing ldapauth_new.c.

The calls are in the Makefile: 
```
CC = gcc
LN = gcc
```

and could be changed to something like ``CC ?= gcc`` to allow us to easily override from the ebuild.

Inspection of the build log is useful to confirm the issue: ``gcc -fpic -march=native -O2 -pipe -frecord-gcc-switches -c -pthread -DGETHOSTBYNAME_R -D_THREAD_SAFE -D_REENTRANT -DNOODBC -DWITH_STD_MALLOC -DFD_SETSIZE=4096 -DWITH_POLL -c -pthread -DGETHOSTBYNAME_R -D_THREAD_SAFE -D_REENTRANT -DNOODBC -DWITH_STD_MALLOC -DFD_SETSIZE=4096 -DWITH_POLL -c -pthread -DGETHOSTBYNAME_R -D_THREAD_SAFE -D_REENTRANT -DNOODBC -DWITH_STD_MALLOC -DFD_SETSIZE=4096 -DWITH_POLL pcre_compile.c``

**Pull request closed without merging**